packages/shared/src/types/PostCollectionConfiguration.ts

```diff
@@ -0,0 +1,166 @@
+import * as db from '../db';
+import * as ChannelAction from './ChannelActions';
+
+/**
+ * How do we want to lay out a collection of posts, at a high level?
+ */
+export type PostCollectionLayoutType =
+  | 'comfy-list-top-to-bottom' // think: notebook
+  | 'compact-list-bottom-to-top' // think: chat
+  | 'grid'; // think: gallery
+
+// Why overload this function instead of just doing a union?
+// If the caller has a non-nullable `channel`, they can then get a
+// non-nullable return value - nice, right?
+export function postCollectionLayoutTypeFromChannel(
+  channel: db.Channel
+): PostCollectionLayoutType;
+export function postCollectionLayoutTypeFromChannel(
+  channel: db.Channel | null
+): PostCollectionLayoutType | null;
+export function postCollectionLayoutTypeFromChannel(
+  channel: db.Channel | null
+): PostCollectionLayoutType | null {
+  switch (channel?.type) {
+    case null:
+    // fallthrough
+    case undefined:
+      return null;
+
+    case 'chat':
+    // fallthrough
+    case 'dm':
+    // fallthrough
+    case 'groupDm':
+      return 'compact-list-bottom-to-top';
+
+    case 'notebook':
+      return 'comfy-list-top-to-bottom';
+
+    case 'gallery':
+      return 'grid';
+  }
+}
+
+/**
+ * Features of a given `PostCollectionLayoutType` - these should be constant
+ * w.r.t. the layout type (i.e. not change based on whether the channel is a DM
+ * vs chat).
+ * If you need to branch based on the channel type, try using
+ * `PostCollectionConfiguration` instead.
+ */
+export interface PostCollectionLayout {
+  columnCount: 1 | 2;
+
+  /** if true, enables day / unread dividers between elements */
+  dividersEnabled: boolean;
+
+  /**
+   * If true, entering channel scrolls to the viewer's first unread post.
+   */
+  enableUnreadAnchor: boolean;
+
+  /**
+   * Width/height ratio for a collection element; e.g. 1 for square, 2 for
+   * landscape, 0.5 for portrait. If null, defers to item sizing.
+   */
+  itemAspectRatio: number | null;
+
+  /**
+   * When new content comes in, should we try to keep the user's scroll position?
+   */
+  shouldMaintainVisibleContentPosition: boolean;
+
+  scrollDirection: 'bottom-to-top' | 'top-to-bottom';
+}
+
+export function postCollectionLayoutForType(
+  layoutType: PostCollectionLayoutType
+): PostCollectionLayout {
+  switch (layoutType) {
+    case 'compact-list-bottom-to-top':
```

Comment:
it's not clear why we would tie the layout to the config. seems like instead you'd want to switch on channel type?

Reply:
what about a custom channel, like an instagram clone which would have a notebook-like layout (i.e. `comfy-list-top-to-bottom`) but with gallery-like inputs? I don't think we should set `channel.type = 'notebook'` there – we could use `channel.type = 'custom'` or `channel.type = 'tlon.r0.instaclone'` and do deeper custom logic, but abstracting the layout into a set of predefined types seems the best mix of simplicity + extensibility for now. wdyt?

Reply:
gonna leave this open for 15 min (until 1:30pm PT) in case you had a blocking response to this, but also happy to follow up

Reply:
np yeah that makes sense, how will we map custom channel types to layout types?